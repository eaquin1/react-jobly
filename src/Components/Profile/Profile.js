import React, {useState, useContext} from "react";
import {Label, Form, Input, FormGroup, Container, Button} from "reactstrap"
import UserContext from "../../UserContext"
import JoblyApi from "../../Helpers/JoblyApi"
import {useHistory} from "react-router-dom"

function Profile() {
    const {currentUser} = useContext(UserContext)
    const history= useHistory();
    
    //get info on user, to fill out the form
    const INITIAL_STATE = {
        first_name: currentUser.first_name|| undefined,
        last_name: currentUser.last_name || undefined,
        email: currentUser.email || undefined,
        photo_url: currentUser.photo_url || undefined,
        password: undefined
    }
    const [formData, setFormData] = useState(INITIAL_STATE)

    const handleChange = (e) => {
        const {name, value} = e.target;
        setFormData(data => ({
            ...data, [name]: value
        }))
    }

    const handleSubmit = async (e) => {
        e.preventDefault();
        await JoblyApi.updateUser(currentUser.username, formData);

        history.push("/jobs")

    }
    return (
        <Container>
        <h1>Profile</h1>
        <Form onSubmit={handleSubmit}>
        <Label for="username">Username</Label>
        <p>{currentUser.username}</p>
        <FormGroup>
            <Label for="first_name">First Name</Label>
                <Input 
                type="text"
                name="first_name"
                value={formData.first_name}
                onChange={handleChange}
                />
            
        </FormGroup>
        <FormGroup>
            <Label>Last Name</Label>
                <Input 
                type="text"
                name="last_name"
                value={formData.last_name}
                onChange={handleChange}
                />
        </FormGroup>
        <FormGroup>
            <Label>Email</Label>
                <Input 
                type="text"
                name="email"
                value={formData.email}
                onChange={handleChange}
                />    
        </FormGroup>
        <FormGroup>
            <Label>Photo URL</Label>
                <Input 
                type="text"
                name="photo_url"
                value={formData.photo_url}
                onChange={handleChange}
                />
            
        </FormGroup>
        <FormGroup>
            <Label>Re-enter Password</Label>
                <Input 
                type="password"
                name="password"
                value={formData.password}
                onChange={handleChange}
                />   
        </FormGroup>
        <Button>Save Changes</Button>
        </Form>
        </Container>

    )
    }

export default Profile