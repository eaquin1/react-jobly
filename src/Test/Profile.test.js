import React from 'react';
import { render } from '@testing-library/react';
import Profile from '../Components/Profile/Profile';
import { UserProvider } from '../testUtils';

describe('Testing the Profile component', () => {
  it('renders without crashing', () => {
    render(
      <UserProvider>
        <Profile />
      </UserProvider>
    );
  });

  it('matches snapshot', () => {
    const { asFragment } = render(
      <UserProvider>
        <Profile />
      </UserProvider>
    );
    expect(asFragment()).toMatchSnapshot();
  });
});