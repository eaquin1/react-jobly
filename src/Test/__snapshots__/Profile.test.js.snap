// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Profile component matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="col-md-6 col-lg-4 offset-md-3 offset-lg-4"
  >
    <h1>
      Profile
    </h1>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <form
          class=""
        >
          <label
            class=""
            for="username"
          >
            Username
          </label>
          <p>
            test
          </p>
          <div
            class="form-group"
          >
            <label
              class=""
              for="first_name"
            >
              First Name
            </label>
            <input
              class="form-control"
              name="first_name"
              type="text"
              value="test"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class=""
            >
              Last Name
            </label>
            <input
              class="form-control"
              name="last_name"
              type="text"
              value="mctest"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class=""
            >
              Email
            </label>
            <input
              class="form-control"
              name="email"
              type="text"
              value="test@test.net"
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class=""
            >
              Photo URL
            </label>
            <input
              class="form-control"
              name="photo_url"
              type="text"
              value=""
            />
          </div>
          <div
            class="form-group"
          >
            <label
              class=""
            >
              Re-enter Password
            </label>
            <input
              class="form-control"
              name="password"
              type="password"
              value=""
            />
          </div>
          <button
            class="btn btn-secondary"
          >
            Save Changes
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
